src/test/java/htsjdk/samtools/SamReaderFactoryTest.java

```diff
@@ -284,7 +284,37 @@ public void queryInputResourcePermutation(final SamInputResource resource) throw
         }
         reader.close();
     }
-    
+
+    public class NeverFilePathInpurResource extends PathInputResource {
+        public NeverFilePathInpurResource(Path pathResource) {
+            super(pathResource);
+        }
+
+        @Override
+        public File asFile() {
+            return null;
+        }
+    }
+
+    @Test
+    public void streamingPathBamWithFileIndex() throws IOException {
```

Comment:
Rename test case to something like `checkHasIndexForStreamingPathBamWithFileIndex()`, to make it clearer how it differs from the test case below it.